addons/mail/wizard/mail_compose_message.py

```diff
@@ -298,6 +285,37 @@ def _compute_composition_batch(self):
             res_ids = composer._evaluate_res_ids()
             composer.composition_batch = len(res_ids) > 1 if res_ids else False
 
+    @api.depends('composition_mode', 'parent_id')
+    def _compute_model(self):
+        """ Model can be set from parent or using 'active_model' context key
+        frequently used as the composer is most invoked from list or form
+        views. """
+        for composer in self:
+            if composer.parent_id and composer.composition_mode == 'comment':
+                composer.model = composer.parent_id.model
+            elif not composer.model:
+                composer.model = self.env.context.get('active_model')
+
+    @api.depends('composition_mode', 'parent_id')
+    def _compute_res_ids(self):
+        """ Computation may come from parent in comment mode, if set. It takes
+        the parent message's res_id. Otherwise the composer uses the 'active_ids'
+        context key, unless it is too big to be stored in database. Indeed
+        when invoked for big mailings, 'active_ids' may be a very big list.
+        Support of 'active_ids' when sending is granted in order to not always
+        rely on 'res_ids' field. When 'active_ids' is not present, fallback
+        on 'active_id'. """
+        for composer in self.filtered(lambda composer: not composer.res_ids):
+            if composer.parent_id and composer.composition_mode == 'comment':
+                composer.res_ids = [composer.parent_id.res_id]
+            else:
+                active_res_ids = self._parse_res_ids(self.env.context.get('active_ids'))
+                # beware, field is limited in storage, usage of active_ids in context still required
+                if active_res_ids and len(active_res_ids) <= self._batch_size:
+                    composer.res_ids = self.env.context['active_ids']
+                elif not active_res_ids and self.env.context.get('active_id'):
+                    composer.res_ids = [self.env.context['active_id']]
```

Comment:
Should those be manually stringified to clarify that this is a Char field and not some x_2many?